protocol/statetracker/provider_state_tracker.go

```diff
@@ -104,7 +124,20 @@ func (pst *ProviderStateTracker) LatestBlock() int64 {
 }
 
 func (pst *ProviderStateTracker) GetMaxCuForUser(ctx context.Context, consumerAddress, chainID string, epoch uint64) (maxCu uint64, err error) {
-	return pst.stateQuery.GetMaxCuForUser(ctx, consumerAddress, chainID, epoch)
+	latestBlockTime := pst.EventTracker.getLatestBlockTime()
+	downtimeParams := pst.chainTracker.GetDowntimeParams()
+
+	delay := time.Now().UTC().Sub(latestBlockTime)
+
+	epochDuration := downtimeParams.EpochDuration.Milliseconds()
+	virtualEpoch := uint64(0)
+	// check if emergency mode is enabled
+	if delay > downtimeParams.DowntimeDuration {
+		// division delay by epoch duration rounded up, subtract one to skip regular epoch
+		virtualEpoch = uint64((delay.Milliseconds()+epochDuration-1)/epochDuration - 1)
+	}
+
+	return pst.stateQuery.GetMaxCuForUser(ctx, consumerAddress, chainID, epoch, virtualEpoch)
```

Comment:
with your current designs we have two mechanisms, one for changing existing cus and one for new cus, it's a more complex design than just saving the virtual epoch and when opening a session and checking cu left just regard it